Реализован сервер, согласно ТЗ №9.

AntonyukV516 opened the pull request:
Проверял Insomnia, но тесты в программе не удаются, При проверке корректно присылаются нужные коды

AntonyukV516 committed:
Реализован сервер, согласно ТЗ №9.
Проверял Insomnia, но тесты в программе не удаются,
При проверке корректно присылаются нужные коды

AntonyukV516 committed:
Реализован сервер, согласно ТЗ №9.
Проверял Insomnia, но тесты в программе не удаются,
При проверке корректно присылаются нужные коды

AntonyukV516 committed:
Реализован сервер, согласно ТЗ №9.
Проверял Insomnia, но тесты в программе не удаются,
При проверке корректно присылаются нужные коды

AntonyukV516 committed:
Merge remote-tracking branch 'origin/sprint_9-solution-http-api' into sprint_9-solution-http-api

AntonyukV516 committed:
Исправлен codeStyle

vlolad commented on src/server/BaseHttpHandler.java:
```diff
@@ -0,0 +1,44 @@
+       String body = requestBody.toString().trim();
+        Gson gson = new Gson();
+        return gson.toJson(body);
+    }
```
Можно добавить отдельные методы, которые бы отправляли базовые коды ошибок (вроде 404, 406 и т.д.)

vlolad commented on src/server/EpicHandler.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
```
Вся логика класса замешана на одном методе handle. Если потом будет обрабатываться ещё запрос по этому эндпоинту, то имплементировать его обработку будет сложно

Надо разбить метод handle, вынести механизмы ответов в отдельные методы класса. В методе handle останется только логика обработки запроса (парсинг строк, http-метода и т.д.)

vlolad commented on src/server/SubtaskHandler.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
```
Аналогично комментарию в хендлере эпика, надо разбить основной метод на несколько внутренних

vlolad commented on src/server/TaskHandler.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
```
Тоже надо разбить хендлер

vlolad commented on test/serverTest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,97 @@
+/*
```
Код закомментирован

vlolad reviewed (changes requested):
Привет

По функционалу всё хорошо, но хендлеры для типов задач надо рефакторить. Один огромный метод handle во всех. Надо разбить на отдельные методы, куда вынести логику обработки тех или иных запросов

Дополнительно тест оказался полностью закомментирован, надо раскомментировать-доработать)

Хорошего дня!

AntonyukV516 committed:
Разбиты handlers, исправлены typeAdapters,
ручное тестирование не вызывает ошибок, бизнес логика работает согласно тз,
но тесты не понятно почему падают, делал как в теории

AntonyukV516 committed:
Исправлен Code Style

vlolad commented on test/serverTest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,109 @@
+
+        try (HttpClient client = HttpClient
+                .newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
```
Не обязательно использовать builder, можно создавать просто с помощью `HttpClient.newHttpClient()`

vlolad commented on test/serverTest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,109 @@
+    @BeforeEach
+    void startServ() throws IOException {
+        this.server = new HttpTaskServer();
+        server.start();
```
Сервер то создается и запускается, но хендлеры в него не регистрируются. Поэтому и вылетает 404 на тестах

vlolad commented on src/server/HttpTaskServer.java:
```diff
@@ -0,0 +1,46 @@
+import java.net.InetSocketAddress;
+
+public class HttpTaskServer {
+    static TaskMeneger taskMeneger = Managers.getDefault();
```
Поля должны быть приватные и без static. Но желательно final

vlolad commented on src/server/HttpTaskServer.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public static void main(String[] args) throws IOException {
+        httpTaskServer.createContext("/tasks", new TaskHandler(taskMeneger));
```
Для решения проблем с тестами можно из main скопировать кусок кода с регистрацией хендлеров в конструктор

vlolad reviewed (commented):
Привет!

Тесты надо доправить, сейчас падает с ошибкой. Проблема в том, что хендлеры не регистрируются при создании в тесте сервера. Написал, как можно поправить

В остальном ещё одно мелкое замечание по серверу, а так отлично

Хорошего вечера!

AntonyukV516 committed:
Доработаны тесты
В HttpTaskServer не понимаю как сделать поля не статичными,
так чтобы main работал, в текущем состоянии работает с тестов,
 и при ручном тестировании.
 тестов не много, так как логика для каждого типа задач одинаковая

vlolad reviewed (approved):
Привет!

Тесты починены, всё остальное отлично. Принято (￣^￣)ゞ

Поздравляю с завершением первого большого проекта! Дальше будет в чём-то проще, так как "низкоуровневые" задачи вроде работы с запросами возьмет на себя фреймворк. Но скучно точно не будет с:

Хорошего дня!
